### PR TITLE
Spruce up the Readme

### PR DESCRIPTION
**Pull request by CodaFi**
Part 3 of #118

### CODE REVIEW

**Comment by CodaFi** [comment]
Needs more examples in general.

Core:

- Either
- Combinators
- Sections

Swiftz:

- Basically everything.

File: README.md [hunk 1]
```diff
@@ -5,154 +5,118 @@ Swiftz
 
 Swiftz is a Swift library for functional programming.
 
-It defines purely functional data structures and functions.
+It defines functional data structures, functions, idioms, and extensions that augment 
+the Swift standard library.
```
Comment (CodaFi) [inline README.md]: Let's be mutually recursive here and mention Swiftx.

Comment (pthariensflame) [inline README.md]: Yay!

**Comment by CodaFi** [comment]
This is the final issue keeping us from 0.1.0.  Let's knock it out and put out a good clean release.

File: README.md [hunk 2]
```diff
@@ -5,154 +5,85 @@ Swiftz
...
+[Prelude/Base](https://hackage.haskell.org/package/base), [SML
+Basis](http://sml-family.org/Basis/), and the [OCaml Standard
+Library](http://caml.inria.fr/pub/docs/manual-ocaml/stdlib.html).  Elements of
+the library rely on their combinatorial semantics to allow declarative ideas to
+be expressed more clearly in Swift.
+
+Swiftz is a proper superset of [Swiftx](https://github.com/typelift/Swiftx) that
+implements higher-level data types like Lenses, Zippers, HLists, and a number of
+typeclasses integral to programming with the maximum amount of support from the
+type system.  A small number of examples are included in the test suite to
```
Comment (CodaFi) [inline README.md]: Should I reprint these examples here?

Comment (pthariensflame) [inline README.md]: I think so; that way, anyone who wants to look at them doesn't have to go digging through the codebase.  The downside is that they'll have to be updated in synchrony.

Comment (CodaFi) [inline README.md]: Do you have anymore suggestions for examples?   We don't have anything related to Zippers, Concurrency, the Lens Family (besides a few small methods), and the typeclasses

Comment (pthariensflame) [inline README.md]: I'll take a closer look at what we have right now, and see if we can improve it.

**Comment by CodaFi** [comment]
For now, I think I've written enough examples.

Ready for review.

**Comment by pthariensflame** [comment]
Tell me when you think you've gotten all the grammar stuff done.  :P

**Comment by CodaFi** [comment]
I think it's good now. 

Fingers crossed

**Comment by pthariensflame** [comment]
This looks fine to me!  I still have to get around to writing the in-code docs for lenses, though.  ;/

**Comment by CodaFi** [comment]
Onward to v0.1.0!

:sail:

**Pull request merged**